### PR TITLE
Chore/393

### PR DESCRIPTION
**Pull request by ckiss**
[ref](https://github.com/project-vagabond/covid-green-app/issues/393)

### CODE REVIEW

File: src/components/atoms/navbar.tsx [hunk 1]
```diff
@@ -87,7 +90,11 @@ export const NavBar: FC<NavBarProps> = ({
             <TouchableWithoutFeedback
               accessibilityRole="button"
               accessibilityHint={t('navbar:backHint')}
-              onPress={() => navigation.goBack()}>
+              onPress={() => {
+                route && route.name === ScreenNames.CloseContactAlert
```
Comment (floridemai) [inline src/components/atoms/navbar.tsx:94]: @ckiss Should we have the condition "hasAlert" in here? 🤔 I think so.

**Comment by AlanSl** [comment]
This works with the in-app back button, but I suspect if we do this like this we'll get a new pair of issues posted very soon that the Android device back button and also iOS back swipe gesture don't also do this, and take the user to a different screen to the in-app back button.

Does it work to use `navigation.reset` as the navigation action on the alert card onPress, reseting navigation to a state where the My COVID Alert root is the previous page in the history? 

That would fix this more cleanly for the in-app button and the device back actions.

File: src/components/atoms/navbar.tsx (hunk 1 above)
Comment (AlanSl) [reply src/components/atoms/navbar.tsx:94]: Is it possible for the current screen to be `ScreenNames.CloseContactAlert` if `hasAlert` is not true?

Comment (floridemai) [reply src/components/atoms/navbar.tsx:94]: I don’t think so. Don’t mind me, was thinking  about something else.

**Comment by ckiss** [comment]
@AlanSl good point, let me try do that!

**Comment by ckiss** [comment]
@AlanSl @floridemai solved using the `navigation.reset`

**Review by AlanSl — COMMENTED** [review]
Works very nicely for me, including on navigating to the screen from the exposure notification from app closed.

It's not the implementation I expected but I can't see any harm in doing it this way on a useEffect cleanup; however I'd like @floridemai to look at it too before approve and merge in case there's  some case I missed.

**Comment by floridemai** [comment]
It's not working for me: back brings me back to the previous screen. Hopefully I get to ticket right and it should go to My Covid Alerts. [RPReplay_Final1599815114.MP4.zip](https://github.com/project-vagabond/covid-green-app/files/5207897/RPReplay_Final1599815114.MP4.zip).

I think a solution whee you push my covid alert and covid alert screen (in that order) is something to try at least.

**Comment by AlanSl** [comment]
In that case it should be more robust to have the `navigation.reset` as the navigation _to_ the screen (on the alert card and on handling the notification), rather than in the screen's useEffect cleanup after navigating to the screen

**Comment by ckiss** [comment]
@AlanSl I debugged this with Paul, looks like an iOS specific issue with the `navigation.reset`. We decided to keep my initial implementation for ios which fixes it on using the in app back button(the native slide will work too in most of the cases) and the useEffect implementation together. We will come back to this when we have more time to dig deeper.

**Review by floridemai — APPROVED** [review]

**Pull request merged**